docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,730 @@ sidebar: lb4_sidebar
+
+  @property({
+    type: 'string',
+  })
+  content?: string;
+
+  constructor(data?: Partial<Note>) {
+    super(data);
+  }
+}
+
+export interface NoteRelations {
+  // describe navigational properties here
+}
+
+export type NoteWithRelations = Note & NoteRelations;
+```
+
+#### Adjusting The Model File To Use AddCategoryPropertyMixin
+
+The model file only requires a few adjustments:
+
+- import the `BaseEntity` class
+- import the `AddCategoryPropertyMixin` mixin
+- Change the class declaration of `Note` so that it extends the class returned
+  from the mixin function which takes in the `BaseEntity` superclass an input
+
+{% include code-caption.html content="src/models/note.model.ts" %}
+
+```ts
+import {model, property} from '@loopback/repository';
+import {AddCategoryPropertyMixin} from '../mixins/category-property-mixin';
+import {BaseEntity} from './base-entity';
+
+@model()
+export class Note extends AddCategoryPropertyMixin(BaseEntity) {
+  @property({
+    type: 'number',
+    id: true,
+    generated: true,
+  })
+  id?: number;
+
+  @property({
+    type: 'string',
+    required: true,
+  })
+  title: string;
+
+  @property({
+    type: 'string',
+  })
+  content?: string;
+
+  constructor(data?: Partial<Note>) {
+    super(data);
+  }
+}
+
+export interface NoteRelations {
+  // describe navigational properties here
+}
+
+export type NoteWithRelations = Note & NoteRelations;
+```
+
+The required property `category` has now been added to the `Note` model via a
+mixin class factory function.
+
+## Creating A Custom Model Method And Remote Model Method Mixin
+
+This section covers the approach that LoopBack 3 can use to add a custom method
+/ remote method to a model via a mixin, and similarly how LoopBack 4 can add a
+custom method to a repository and controller via a mixin.
+
+### LoopBack 3 Approach
+
+The
+[Add a New Model Method And a New Endpoint](./methods.md#add-a-new-model-method-and-a-new-endpoint)
+section of the [Migrating custom model methods](./methods.md) document explains
+how a LoopBack 3 developer can define a custom model method named `findByTitle`
+on the `Note` model, and define a remote method to make it available as a new
+endpoint.
+
+In this section, we will show how a LoopBack 3 developer can define a mixin to
+accomplish this.
+
+In LoopBack 3, a developer is able to create a custom model method/remote method
+mixin by:
+
+- placing the mixin logic in a file in a mixins directory
+- updating the server/model-config.json file with the mixin directory location
+- updating the model's json file to include the mixin's name (and options object
+  or boolean)
+
+#### Defining The Model Method Mixin findByTitle.js
+
+The developer defines a custom model method/remote method mixin in
+**common/mixins/findByTitle.js** which adds a custom method `findByTitle` to any
+model, and adds a corresponding remote method definition with path
+`/findByTitle` as well. An options property `returnArgumentName` makes it
```

Comment:
```suggestion
`/findByTitle` as well. An option property `returnArgumentName` makes it
```
ignore it if I am wrong :p

Reply:
@jannyHou  : `options` is the name of the parameter. And `returnArgumentName` is a property inside it.